src/NServiceBus.Core/Transports/ErrorContext.cs

```diff
@@ -59,6 +67,6 @@ public ErrorContext(Exception exception, Dictionary<string, string> headers, str
         /// <summary>
         /// A collection of additional information provided by the transport.
         /// </summary>
-        public ReadOnlyContextBag Extensions { get; }
+        public ContextBag Extensions { get; }
```

Comment:
Given that the context needs to be propagated from message handling to error handling [for satellite receivers](https://github.com/Particular/NServiceBus/pull/5937), and [satellite receivers don't receive the receive completed event](https://github.com/Particular/NServiceBus/pull/5936/files#r579799291), I don't understand why they need to be able to write information into the context during error handling.

Reply:
>satellite receivers don't receive the receive completed event

That is just because we haven't exposed it yet. Since context can flow from OnMessage to OnError I'd say its expected for it to be able to flow to OnComplete as well